docs/code/IDataViewTypeSystem.md

```diff
@@ -0,0 +1,843 @@
+  converted in both directions.
+
+## Vector Types
+
+### Introduction
+
+Vector types are one of the key innovations of the IDataView system and are
+critical for high dimensional machine-learning applications.
+
+For example, when processing text, it is common to hash all or parts of the
+text and encode the resulting hash values, first as a key type, then as
+indicator or bag vectors using the `KeyToVector` transform. Using a `k`-bit
+hash produces a key type with `Count` equal to `2^^k`, and vectors of the same
+length. It is common to use `20` or more hash bits, producing vectors of
+length a million or more. The vectors are typically very sparse. In systems
+that do not support vector-valued columns, each of these million or more
+values is placed in a separate (sparse) column, leading to a massive explosion
+of the column space. Most tabular systems are not designed to scale to
+millions of columns, and the user experience also suffers when displaying such
+data. Moreover, since the vectors are very sparse, placing each value in its
+own column means that, when a row is being processed, each of those sparse
+columns must be queried or scanned for its current value. Effectively the
+sparse matrix of values has been needlessly transposed. This is very
+inefficient when there are just a few (often one) non-zero entries among the
+column values. Vector types solve these issues.
+
+A vector type is an instance of the sealed `VectorType` class, which derives
+from `ColumnType`. The vector type contains its `ItemType`, which must be a
+`PrimitiveType`, and its dimensionality information. The dimensionality
+information consists of one or more non-negative integer values. The
+`VectorSize` is the product of the dimensions. A dimension value of zero means
+that the true value of that dimension can vary from value to value.
+
+For example, tokenizing a text by splitting it into multiple terms generates a
+vector of text of varying/unknown length. The result type shorthand is
+`V<TX,*>`. Hashing this using `6` bits then produces the vector type
+`V<U4[0-63],*>`. Applying the `KeyToVector` transform then produces the vector
+type `V<R4,*,64>`. Each of these vector types has a `VectorSize` of zero,
+indicating that the total number of slots varies, but the latter still has
+potentially useful dimensionality information: the vector slots are
+partitioned into an unknown number of runs of consecutive slots each of length
+`64`.
+
+As another example, consider an image data set. The data starts with a `TX`
+column containing URLs for images. Applying a BitmapLoader transform generates
+a column of a custom (non-standard) type, `Picture<*,*,4>`, where the
+asterisks indicate that the picture dimensions are unknown. The last dimension
+of `4` indicates that there are four channels in each pixel: the three color
+components, plus the alpha channel. Applying a `BitmapScaler` transform scales
```

Comment:
I think we deprecated `BitmapScaler` for `ImageResizer` (and similar for the rest of the image transforms.

Reply:
Ah you're right... and image loader transform and so on, and so on.

---
In reply to: [189171796](https://github.com/dotnet/machinelearning/pull/173#discussion_r189171796) [](ancestors = 189171796)